services/brig/src/Brig/User/API/Auth.hs

```diff
@@ -260,13 +262,18 @@ renew ::
   Maybe (Either ZAuth.UserToken ZAuth.LegalHoldUserToken) ->
   Maybe (Either ZAuth.AccessToken ZAuth.LegalHoldAccessToken) ->
   Handler (Either (Auth.Access ZAuth.User) (Auth.Access ZAuth.LegalHoldUser))
-renew (Just (Left ut)) (Just (Left at)) = Left <$> Auth.renewAccess ut (Just at) !>> zauthError
-renew (Just (Left ut)) Nothing = Left <$> Auth.renewAccess @ZAuth.User @ZAuth.Access ut Nothing !>> zauthError
-renew (Just (Right lut)) (Just (Right lat)) = Right <$> Auth.renewAccess @ZAuth.LegalHoldUser @ZAuth.LegalHoldAccess lut (Just lat) !>> zauthError
-renew (Just (Right lut)) Nothing = Right <$> Auth.renewAccess @ZAuth.LegalHoldUser @ZAuth.LegalHoldAccess lut Nothing !>> zauthError
-renew (Just (Left _)) (Just (Right _)) = throwStd authTokenMismatch
-renew (Just (Right _)) (Just (Left _)) = throwStd authTokenMismatch
-renew Nothing _ = throwStd authMissingCookie
+renew = \case
```

Comment:
Not sure if this really makes the code clearer, but I found it hard to follow the pattern matching and it was a nice challenge on a Friday afternoon. :)

Reply:
Perhaps you can write 2-3 lines about what this does in English?  Should be a nice challenge, too.  :)P

Reply:
I added some comments now.

But my this change didn't make it easier to understand the code, maybe I should leave it at the old version, which was verbose, but more explicit, and just add comments to that?

Reply:
no, please keep it, now that it's not quite as late in the day and there is english, it's pretty!